eng/packaging.targets

```diff
@@ -127,4 +127,48 @@
     </ItemGroup>
   </Target>
 
+  <!-- Include a netstandard compat error if the project targets both .NETStandard and
+       .NETCoreApp. This prohibits users to consume packages on an older .NETCorAapp version
+       than the minimum supported one. -->
+  <ItemGroup Condition="'$(DisableNETStandardCompatErrorForNETCoreApp)' != 'true'">
+    <NETStandardCompatError Include="netcoreapp2.0"
```

Comment:
how would this work after we dead end netcoreapp2.1 or netcoreapp3.1 ? would we have to change it ? can we store it in a property which points to the last netcoreapp framework that went out of support.

Reply:
netcoreapp2.1 is already out-of-support and not included in packages. The Include defines the lower boundary for which to throw and the `Supported` attribute the upper exclusive boundary. In this case an error is thrown for `netcoreapp2.0 - netcoreapp3.0`. On `netcoreapp3.1` (= the `$(NetCoreAppMinimum)`) no exception is thrown.

Example of the System.Drawing.Common nuget package:

![image](https://user-images.githubusercontent.com/7412651/128748602-d33e9a99-2d8c-40aa-a1c3-824501ca6e0a.png)

This means that as long as the `$(NetCoreAppMinimum)` property marks the minimum supported version, this logic doesn't need to be updated. For NET7 we might want to add another item for NETFramework as net461 will be out of support by when NET7 ships.